core/trino-main/src/main/java/io/trino/sql/planner/optimizations/DeterminePartitionCount.java

```diff
@@ -108,11 +110,9 @@ public PlanNode optimize(
         requireNonNull(types, "types is null");
         requireNonNull(tableStatsProvider, "tableStatsProvider is null");
 
-        // Skip for write nodes since writing partitioned data with small amount of nodes could cause
-        // memory related issues even when the amount of data is small. Additionally, skip for FTE mode since we
-        // are not using estimated partitionCount in FTE scheduler.
-        if (PlanNodeSearcher.searchFrom(plan).whereIsInstanceOfAny(INSERT_NODES).matches()
-                || getRetryPolicy(session) == RetryPolicy.TASK) {
+        // For streaming execution, skip for write nodes since writing partitioned data with small amount of nodes could cause
+        // memory related issues even when the amount of data is small.
+        if (!getRetryPolicy(session).equals(RetryPolicy.TASK) && PlanNodeSearcher.searchFrom(plan).whereIsInstanceOfAny(INSERT_NODES).matches()) {
             return plan;
         }
```

Comment:
Adaptive partitioning is disabled for writes in streaming execution, is this a problem in FTE as well? @arhimondr ?

Reply:
Yes - it makes sense to keep lots of partitons if we are writing to avoid "too many open writers" problem. Not if 1000 is needed though. If so it would be super rare.